Mathlib/CategoryTheory/Adjunction/FullyFaithful.lean

```diff
@@ -215,4 +214,20 @@ def Adjunction.restrictFullyFaithful (iC : C ⥤ C') (iD : D ⥤ D') {L' : C' 
         apply comm2.hom.naturality g }
 #align category_theory.adjunction.restrict_fully_faithful CategoryTheory.Adjunction.restrictFullyFaithful
 
+instance [L.Faithful] [L.Full] {Y : C} : IsIso (h.counit.app (L.obj Y)) :=
+  isIso_of_hom_comp_eq_id _ (h.left_triangle_components Y)
+
+lemma isIso_counit_app_of_iso [L.Faithful] [L.Full] (X : D) (Y : C) (e : X ≅ L.obj Y) :
+    IsIso (h.counit.app X) := by
+  rw [NatTrans.isIso_app_iff_of_iso _ e]
+  infer_instance
+
```

Comment:
Maybe make it into an iff?
Something like `IsIso (h.counit.app X) \iff X \in L.essImage`.
Also I think these should go before `Adjunction.restrictFullyFaithful` (that def doesn't belong to this file IMO).

Reply:
#12363 moves the def to a new file